apps/web/hooks/useSubgraphQueryMultiChain.ts

```diff
@@ -0,0 +1,147 @@
+  variables: Variables = {} as Variables,
+  context?: Partial<OperationContext>,
+  changeScope?: ChangeEventScope[] | ChangeEventScope,
+) {
+  const { connected, subscribe, unsubscribe } = usePubSubContext();
+
+  const [response, setResponse] = useState<Data[]>();
+  const [fetching, setFetching] = useState(true);
+
+  const responseMap = useRef(new Map<ChainId, Data>());
+  const errorsMap = useRef(new Map<ChainId, CombinedError>());
+  const subscritionId = useRef<SubscriptionId>();
+
+  useEffect(() => {
+    const init = async () => {
+      setFetching(true);
+      await fetchDebounce();
+    };
+    init();
+  }, []);
+
+  useEffect(() => {
+    if (!connected || !changeScope || changeScope.length === 0) {
+      return;
+    }
+
+    subscritionId.current = subscribe(changeScope, (payload) => {
+      fetchDebounce(payload.chainId ? [payload.chainId] : undefined, true);
+    });
+
+    return () => {
+      if (subscritionId.current) {
+        unsubscribe(subscritionId.current);
+      }
+    };
+  }, [connected]);
+
+  const fetchDebounce = debounce(
+    async (chainsOverride?: ChainId[], retryOnNoChange?: boolean) => {
+      const chainSubgraphs = (chainsOverride ?? allChains).map((chain) => ({
+        chainId: chain,
+        url: getContractsAddrByChain(chain)?.subgraphUrl,
+      }));
+      await Promise.all(
+        chainSubgraphs.map(async ({ chainId, url }, i) => {
+          const fetchSubgraphChain = async (retryCount?: number) => {
+            try {
+              const fetchQuery = async () => {
+                const { urqlClient } = initUrqlClient({
+                  chainId: (chainsOverride ?? allChains)[i],
+                });
+                return await urqlClient.query<Data>(query, variables, {
+                  ...context,
+                  url,
+                  chainId,
+                  requestPolicy: "network-only",
+                } as OperationContext & { _instance: any });
+              };
+
+              const res = await fetchQuery();
+
+              if (res.error) {
+                errorsMap.current.set(chainId, res.error);
+              } else {
+                if (
+                  !isEqual(res.data, responseMap.current.get(chainId)) ||
+                  retryCount === undefined ||
+                  retryCount >= CHANGE_EVENT_MAX_RETRIES
+                ) {
+                  if (retryCount === CHANGE_EVENT_MAX_RETRIES) {
+                    console.debug(
+                      `Still not updated but max retries reached. (retry count: ${retryCount})`,
+                    );
+                  }
+                  responseMap.current.set(chainId, res.data!);
+                } else {
+                  console.debug(
+                    `Subgraph-${chainId} result not yet updated, retrying with incremental delays... (retry count: ${retryCount + 1}/${CHANGE_EVENT_MAX_RETRIES})`,
+                  );
+                  const delay = 2000 * 2 ** retryCount;
```

Comment:
should be in CONSTANT the delay?

Reply:
yeah only the CHANGE_EVENT_INITIAL_DELAY

Reply:
https://github.com/1Hive/gardens-v2/pull/229/commits/36d5d5c6e42e074f66dfb1030edfe1d7d80665db